app/celery/celery.py

```diff
@@ -58,6 +57,11 @@ def apply_async(self, args=None, kwargs=None, task_id=None, producer=None,
 
             if has_request_context() and hasattr(request, 'request_id'):
                 kwargs['request_id'] = request.request_id
+            else:
```

Comment:
Optional: What you've written is all good I think it looks a bit cleaner as the following

```
elif g.hasattr("request_id") and g.request_id:
    kwargs['request_id'] = g.request_id
```

But as I said, all fine so you can pick what you like most

Reply:
g doesn't have hasattr(), but I've taken the same approach as notifications-utils